src/test/java/com/powsybl/openloadflow/sa/OpenSecurityAnalysisWithActionsTest.java

```diff
@@ -869,75 +869,75 @@ private void testGeneratorAction(boolean dc, LoadFlowParameters.BalanceType bala
         parameters.setDistributedSlack(true).setBalanceType(balanceType);
         parameters.setDc(dc);
         OpenLoadFlowParameters openLoadFlowParameters = new OpenLoadFlowParameters();
-        openLoadFlowParameters.setSlackBusPMaxMismatch(slackBusPMaxMismatch);
+        openLoadFlowParameters.setSlackBusPMaxMismatch(1e-3);
         parameters.addExtension(OpenLoadFlowParameters.class, openLoadFlowParameters);
         SecurityAnalysisParameters securityAnalysisParameters = new SecurityAnalysisParameters();
         securityAnalysisParameters.setLoadFlowParameters(parameters);
 
         SecurityAnalysisResult result = runSecurityAnalysis(network, contingencies, monitors, securityAnalysisParameters,
                 operatorStrategies, actions, ReportNode.NO_OP);
 
+        double deltaPowerAssert = 5e-3; // We need a larger delta power assert since we distribute some power in slack distribution steps
+
         // verify security analysis result through load flows step by step
         // apply contingency
         network.getLine(lineInContingencyId).getTerminal1().disconnect();
         network.getLine(lineInContingencyId).getTerminal2().disconnect();
         loadFlowRunner.run(network, parameters);
-        assertEquals(network.getLine("l12").getTerminal1().getP(), getPostContingencyResult(result, lineInContingencyId).getNetworkResult().getBranchResult("l12").getP1(), LoadFlowAssert.DELTA_POWER);
-        assertEquals(network.getLine("l14").getTerminal1().getP(), getPostContingencyResult(result, lineInContingencyId).getNetworkResult().getBranchResult("l14").getP1(), LoadFlowAssert.DELTA_POWER);
-        assertEquals(network.getLine("l23").getTerminal1().getP(), getPostContingencyResult(result, lineInContingencyId).getNetworkResult().getBranchResult("l23").getP1(), LoadFlowAssert.DELTA_POWER);
-        assertEquals(network.getLine("l34").getTerminal1().getP(), getPostContingencyResult(result, lineInContingencyId).getNetworkResult().getBranchResult("l34").getP1(), LoadFlowAssert.DELTA_POWER);
+        assertEquals(network.getLine("l12").getTerminal1().getP(), getPostContingencyResult(result, lineInContingencyId).getNetworkResult().getBranchResult("l12").getP1(), deltaPowerAssert);
+        assertEquals(network.getLine("l14").getTerminal1().getP(), getPostContingencyResult(result, lineInContingencyId).getNetworkResult().getBranchResult("l14").getP1(), deltaPowerAssert);
+        assertEquals(network.getLine("l23").getTerminal1().getP(), getPostContingencyResult(result, lineInContingencyId).getNetworkResult().getBranchResult("l23").getP1(), deltaPowerAssert);
+        assertEquals(network.getLine("l34").getTerminal1().getP(), getPostContingencyResult(result, lineInContingencyId).getNetworkResult().getBranchResult("l34").getP1(), deltaPowerAssert);
 
```

Comment:
@vmouradian why do you need a larger tolerance here ? The comparison is between the security analysis engine and the manual disconnection. The results should be close an independant of the change made on the generatorActions.

Reply:
Hello @vidaldid-rte,  I did this modification, not @vmouradian.
We need a larger tolerance since the computations are not exactly the same.

Indeed, with the manual disconnection, you run only one loadflow with different NRs separated by the slack distribution steps.
With the generator action, you run two loadflows, one before and one after action application. Each load flow has its own slack distribution steps.

So, in theory you should have exactly same results, but in practice, there is a stopping criteria on the loadflow algorithm, using the slackBusPMaxMismatch parameter. Since some non-negligeable power is distributed in those tests, depending on the value of this parameter, you end up with some differences between manual disconnection application and action application results.

Reply:
@Hadrien-Godard 
I believe that the issue is the stopping criteria. I see no reason why the compensations would be different.
See my comment below in changing the stopping criteria.

You can as well use that one instead. It remains a global criteria on the sum of errors but wit the size of the test network prevent an error larger than 1e-3 to accumulate on a single equation.
        openLoadFlowParameters.setNewtonRaphsonConvEpsPerEq(1e-5);

Reply:
It seems to work well with 
openLoadFlowParameters.setNewtonRaphsonStoppingCriteriaType(NewtonRaphsonStoppingCriteriaType.PER_EQUATION_TYPE_CRITERIA);
openLoadFlowParameters.setMaxActivePowerMismatch(1e-3);

I push the modification.